### PR TITLE
Fix: check if addon is enabled before doing validation

### PR DESCRIPTION
**Pull request by kadern0**
Fixes: #15432, #15429, #15431, #15430

```
$ ./out/minikube addons list
|-----------------------------|----------|--------------|--------------------------------|
|         ADDON NAME          | PROFILE  |    STATUS    |           MAINTAINER           |
|-----------------------------|----------|--------------|--------------------------------|
| ambassador                  | minikube | disabled     | 3rd party (Ambassador)         |
| auto-pause                  | minikube | disabled     | Google                         |
| cloud-spanner               | minikube | disabled     | Google                         |
| csi-hostpath-driver         | minikube | disabled     | Kubernetes                     |
| dashboard                   | minikube | disabled     | Kubernetes                     |
| default-storageclass        | minikube | enabled ✅   | Kubernetes                     |
| efk                         | minikube | disabled     | 3rd party (Elastic)            |
| freshpod                    | minikube | disabled     | Google                         |
| gcp-auth                    | minikube | disabled     | Google                         |
| gvisor                      | minikube | disabled     | Google                         |
| headlamp                    | minikube | disabled     | 3rd party (kinvolk.io)         |
| helm-tiller                 | minikube | disabled     | 3rd party (Helm)               |
| inaccel                     | minikube | disabled     | 3rd party (InAccel             |
|                             |          |              | [info@inaccel.com])            |
| ingress                     | minikube | disabled     | Kubernetes                     |
| ingress-dns                 | minikube | disabled     | Google                         |
| istio                       | minikube | disabled     | 3rd party (Istio)              |
| istio-provisioner           | minikube | disabled     | 3rd party (Istio)              |
| kong                        | minikube | disabled     | 3rd party (Kong HQ)            |
| kubevirt                    | minikube | disabled     | 3rd party (KubeVirt)           |
| logviewer                   | minikube | disabled     | 3rd party (unknown)            |
| metallb                     | minikube | disabled     | 3rd party (MetalLB)            |
| metrics-server              | minikube | disabled     | Kubernetes                     |
| nvidia-driver-installer     | minikube | disabled     | Google                         |
| nvidia-gpu-device-plugin    | minikube | disabled     | 3rd party (Nvidia)             |
| olm                         | minikube | disabled     | 3rd party (Operator Framework) |
| pod-security-policy         | minikube | disabled     | 3rd party (unknown)            |
| portainer                   | minikube | disabled     | 3rd party (Portainer.io)       |
| registry                    | minikube | disabled     | Google                         |
| registry-aliases            | minikube | disabled     | 3rd party (unknown)            |
| registry-creds              | minikube | disabled     | 3rd party (UPMC Enterprises)   |
| storage-provisioner         | minikube | enabled ✅   | Google                         |
| storage-provisioner-gluster | minikube | disabled     | 3rd party (Gluster)            |
| volumesnapshots             | minikube | disabled     | Kubernetes                     |
|-----------------------------|----------|--------------|--------------------------------|
```

Before
```
$ minikube addons disable gvisor

❌  Exiting due to MK_ADDON_DISABLE: run callbacks: running validations: [
This addon can only be enabled with the containerd runtime backend. To enable this backend, please first stop minikube with:

minikube stop

and then start minikube again with the following flags:

minikube start --container-runtime=containerd --docker-opt containerd=/var/run/containerd/containerd.sock]

╭───────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                           │
│    😿  If the above advice does not help, please let us know:                             │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                           │
│                                                                                           │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│    Please also attach the following file to the GitHub issue:                             │
│    - /tmp/minikube_addons_569b151b47196d038af25537206255c050ace6fd_0.log                  │
│                                                                                           │
╰───────────────────────────────────────────────────────────────────────────────────────────╯
```
After
```
$ ./out/minikube addons disable gvisor
🌑  "The 'gvisor' addon is disabled
```
Before:
```
 minikube addons disable gcp-auth

❌  Exiting due to MK_ADDON_DISABLE: run callbacks: running callbacks: [Process exited with status 1]

╭───────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                           │
│    😿  If the above advice does not help, please let us know:                             │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                           │
│                                                                                           │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│    Please also attach the following file to the GitHub issue:                             │
│    - /tmp/minikube_addons_2c4fde7ba5d6af1ee47e28fd9619be21ed484a50_0.log                  │
│                                                                                           │
╰───────────────────────────────────────────────────────────────────────────────────────────╯
```
After:
```
$ ./out/minikube addons disable gcp-auth
🌑  "The 'gcp-auth' addon is disabled
```
Before:
```
$ minikube addons disable testing

❌  Exiting due to MK_ADDON_DISABLE: run callbacks: testing is not a valid addon

╭───────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                           │
│    😿  If the above advice does not help, please let us know:                             │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                           │
│                                                                                           │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│    Please also attach the following file to the GitHub issue:                             │
│    - /tmp/minikube_addons_4755f700bf6c955ce4ad0731b6be6682109661fe_0.log                  │
│                                                                                           │
╰───────────────────────────────────────────────────────────────────────────────────────────╯

```
After:
```
$ ./out/minikube addons disable testing

❌  Exiting due to MK_ADDON_DISABLE: "'testing' is not a valid minikube addon

```

### CODE REVIEW

File: cmd/minikube/cmd/config/disable.go [hunk 1]
```diff
@@ -38,11 +40,24 @@ var addonsDisableCmd = &cobra.Command{
...
+		}
+		validAddon, ok := assets.Addons[addon]
+		if ok {
+			if validAddon.IsEnabled(cc) {
+				err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
+				if err != nil {
+					exit.Error(reason.InternalAddonDisable, "disable failed", err)
+				}
+			} else {
+				out.Step(style.AddonDisable, `"The '{{.minikube_addon}}' addon is disabled`, out.V{"minikube_addon": addon})
```
Comment (spowelljr) [inline cmd/minikube/cmd/config/disable.go:55]: I don't think this should be in an `else`. Before after `addons.SetAndSave` was called, if it was successful it would output `The '{{.minikube_addon}}' addon is disabled`. But now it's only outputting the message if the addon is already disabled.

File: cmd/minikube/cmd/config/disable.go [hunk 2]
```diff
@@ -38,11 +40,24 @@ var addonsDisableCmd = &cobra.Command{
 		if addon == "heapster" {
 			exit.Message(reason.AddonUnsupported, "The heapster addon is depreciated. please try to disable metrics-server instead")
 		}
-		err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
+		cc, err := config.Load(ClusterFlagValue())
 		if err != nil {
-			exit.Error(reason.InternalAddonDisable, "disable failed", err)
+			exit.Error(reason.InternalAddonDisable, "loading profile", err)
```
Comment (medyagh) [inline cmd/minikube/cmd/config/disable.go:45]: the reason should be something different, if there is not a reason for it we can create one

**Review by medyagh — CHANGES_REQUESTED** [review]
do you mind posting before/after this PR ? in the PR description?

File: cmd/minikube/cmd/config/disable.go [hunk 3]
```diff
@@ -38,11 +40,24 @@ var addonsDisableCmd = &cobra.Command{
...
+		}
+		validAddon, ok := assets.Addons[addon]
+		if ok {
```
Comment (spowelljr) [inline cmd/minikube/cmd/config/disable.go:48]: nit: With the change of the non-zero exit code below this `ok` can be inverted and reduce the nesting.

**Before:**
```
if ok {
	if validAddon.IsEnabled(cc) {
		err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
		if err != nil {
			exit.Error(reason.InternalAddonDisable, "disable failed", err)
		}
	} else {
		out.Step(style.AddonDisable, `"The '{{.minikube_addon}}' addon is disabled`, out.V{"minikube_addon": addon})
	}

} else {
	out.Step(style.AddonDisable, `"The '{{.minikube_addon}}' seems not to be a valid minikube addon`, out.V{"minikube_addon": addon})
}
```

**After:**
```
if !ok {
	exit.Message(reason.InternalAddonDisable, `"'{{.minikube_addon}}' is not a valid minikube addon`, out.V{"minikube_addon": addon})
}
if validAddon.IsEnabled(cc) {
	err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
	if err != nil {
		exit.Error(reason.InternalAddonDisable, "disable failed", err)
	}
}
out.Styled(style.AddonDisable, `"The '{{.minikube_addon}}' addon is disabled`, out.V{"minikube_addon": addon})
```

**Review by spowelljr — CHANGES_REQUESTED** [review]
Thanks for the PR, just a few things, also, once you're done making the changes could you also run `make generate-docs` to add the new message to translations, thanks

**Review by spowelljr — CHANGES_REQUESTED** [review]
Could you also add a test for this, just try disabling one of the addons that was throwing and erro and make sure it's a non-zero exit code

File: cmd/minikube/cmd/config/disable.go [hunk 4]
```diff
@@ -38,11 +40,24 @@ var addonsDisableCmd = &cobra.Command{
...
+				err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
+				if err != nil {
+					exit.Error(reason.InternalAddonDisable, "disable failed", err)
+				}
+			} else {
+				out.Step(style.AddonDisable, `"The '{{.minikube_addon}}' addon is disabled`, out.V{"minikube_addon": addon})
+			}
+
+		} else {
+			out.Step(style.AddonDisable, `"The '{{.minikube_addon}}' seems not to be a valid minikube addon`, out.V{"minikube_addon": addon})
```
[the remark this answers is not in the document]
Comment (kadern0) [reply cmd/minikube/cmd/config/disable.go:59]: makes sense

File: cmd/minikube/cmd/config/disable.go (hunk 1 above)
Comment (kadern0) [reply cmd/minikube/cmd/config/disable.go:55]: I think you are correct

File: cmd/minikube/cmd/config/disable.go (hunk 2 above)
Comment (kadern0) [reply cmd/minikube/cmd/config/disable.go:45]: Do we need a reason or just an error message instead?

Comment (medyagh) [reply cmd/minikube/cmd/config/disable.go]: reason package is a way to help users quickly identify the Real Reason minikube failed... so it would generate a unique keyword for the error message
also when users collect metrics what type of failures minikube have...we can identify this as a Unique one...and measure how many times this specific err happens. (so we prefer as many unique error key words (reasons) for uniuqe reaons)

File: cmd/minikube/cmd/config/disable.go [hunk 5]
```diff
@@ -38,11 +40,22 @@ var addonsDisableCmd = &cobra.Command{
 		if addon == "heapster" {
 			exit.Message(reason.AddonUnsupported, "The heapster addon is depreciated. please try to disable metrics-server instead")
 		}
-		err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
+		cc, err := config.Load(ClusterFlagValue())
```
Comment (spowelljr) [inline cmd/minikube/cmd/config/disable.go:43]:
```suggestion
		_, cc := mustload.Partial(ClusterFlagValue())
```
Since you require the `ClusterConfig` I recommend you use `mustload.Partial` instead as that will handle exiting if the cluster can't be loaded.

File: cmd/minikube/cmd/config/disable.go [hunk 6]
```diff
@@ -38,11 +40,22 @@ var addonsDisableCmd = &cobra.Command{
...
 		if err != nil {
-			exit.Error(reason.InternalAddonDisable, "disable failed", err)
+			exit.Error(reason.InternalConfigView, "loading profile", err)
 		}
-		out.Step(style.AddonDisable, `"The '{{.minikube_addon}}' addon is disabled`, out.V{"minikube_addon": addon})
+		validAddon, ok := assets.Addons[addon]
+
+		if !ok {
+			exit.Message(reason.InternalAddonDisable, `"'{{.minikube_addon}}' is not a valid minikube addon`, out.V{"minikube_addon": addon})
```
Comment (spowelljr) [inline cmd/minikube/cmd/config/disable.go:50]:
```suggestion
			exit.Message(reason.AddonUnsupported, `"'{{.minikube_addon}}' is not a valid minikube addon`, out.V{"minikube_addon": addon})
```
I think `AddonUnsupported` makes more sense here as it's not an internal error, but trying to use an invalid addon.

**Review by spowelljr — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/config/disable.go (hunk 2 above)
Comment (kadern0) [reply cmd/minikube/cmd/config/disable.go]: Makes sense. Have a look at the updated code, I think it is good now.

**Review by spowelljr — APPROVED** [review]
Looks good, thanks for the PR!
